odoo/modules/loading.py

```diff
@@ -410,6 +427,10 @@ def load_modules(db, force_demo=False, status=None, update_module=False):
                 cr.commit()
                 return registry
 
+        # STEP 5.5: verify extended fields on every model
+        if models_to_check:
+            registry.init_models(cr, list(models_to_check), {})
```

Comment:
comment:
```python
# This will fix the schema of all models in a situation like:
#  - module A is loaded and defines model M;
#  - module B is installed/upgraded and extends model M;
#  - module C is loaded and extends model M.
# The changes introduced by module C are not taken into account by
# the upgrade of module B.
```